libfwupd/fwupd-enums.h

```diff
@@ -220,6 +220,34 @@ typedef enum {
 	FWUPD_RELEASE_URGENCY_LAST
 } FwupdReleaseUrgency;
 
+/**
+ * FwupdPluginFlags:
+ * @FWUPD_PLUGIN_FLAG_NONE:			No flags set
+ * @FWUPD_PLUGIN_FLAG_DISABLED:			Disabled
+ * @FWUPD_PLUGIN_FLAG_USER_WARNING:		Show the user a warning
+ * @FWUPD_PLUGIN_FLAG_CLEAR_UPDATABLE:		Clear the UPDABLE flag from devices
+ * @FWUPD_PLUGIN_FLAG_NO_HARDWARE:		No hardware is found
+ * @FWUPD_PLUGIN_FLAG_CAPSULES_UNSUPPORTED:	UEFI UpdateCapsule are unsupported
+ * @FWUPD_PLUGIN_FLAG_UNLOCK_REQUIRED:		Hardware unlock is required
+ * @FWUPD_PLUGIN_FLAG_EFIVAR_NOT_MOUNTED:	The efivar filesystem is not found
+ * @FWUPD_PLUGIN_FLAG_ESP_NOT_FOUND:		The EFI ESP not found
+ * @FWUPD_PLUGIN_FLAG_LEGACY_BIOS:		System running in legacy CSM mode
+ *
+ * The plugin flags.
+ **/
+#define FWUPD_PLUGIN_FLAG_NONE			(0u)		/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_DISABLED		(1u << 0)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_USER_WARNING		(1u << 1)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_CLEAR_UPDATABLE	(1u << 2)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_NO_HARDWARE		(1u << 3)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_CAPSULES_UNSUPPORTED	(1u << 4)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_UNLOCK_REQUIRED	(1u << 5)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_EFIVAR_NOT_MOUNTED	(1u << 6)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_ESP_NOT_FOUND		(1u << 7)	/* Since: 1.5.0 */
+#define FWUPD_PLUGIN_FLAG_LEGACY_BIOS		(1u << 8)	/* Since: 1.5.0 */
```

Comment:
I think these need `REASON_DEGRADED` in the name or so.

Reply:
I moved away from degraded for two reasons:

 * It made the define much longer
 * it's not degrading the plugin per-se, but it's just a property (i.e. a thing the plugin has noticed)